packages/ui/checkbox/src/checkbox.tsx

```diff
@@ -1,137 +1,110 @@
-import React, {
-	forwardRef,
-	useCallback,
-	useId,
-	useEffect,
-	useState,
-} from "react"
-import { useInteraction } from "@wpmudev/sui-hooks"
+import React, { useEffect, useId } from "react"
+
 import { generateCN } from "@wpmudev/sui-utils"
-import { Tick } from "./elements/tick"
-import { Indeterminate } from "./elements/indeterminate"
+import { useInteraction } from "@wpmudev/sui-hooks"
 
-import { CheckboxProps } from "./checkbox.types"
+import { Indeterminate } from "./elements/indeterminate"
+import { Tick } from "./elements/tick"
 import { useCheckbox } from "./checkbox-context"
+import { CheckboxProps } from "./checkbox.types"
 
-/**
- * Checkbox
- *
- * A React component used in forms when a user needs to select
- * multiple values from several options.
- *
- * @param {CheckboxProps} props - Props for the Checkbox component.
- * @return {React.FC} The Checkbox component.
- */
-const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
-	(
-		{
-			id,
-			label,
-			isLabelHidden = false,
-			isDisabled = false,
-			isSmall: propIsSmall = false,
-			value = "",
-			isIndeterminate = false,
-			isChecked: propIsChecked = false,
-			onChange: propOnchange = () => {},
-		},
-		ref,
-	) => {
-		// State to manage the checked state of the checkbox
-		const [isChecked, setIsChecked] = useState<boolean>(propIsChecked ?? false)
-
-		const { onChange, name, isSmall } = useCheckbox()
-
-		// Set the initial state based on the default value
-		useEffect(() => {
-			setIsChecked((propIsChecked || isIndeterminate) ?? false)
-		}, [propIsChecked, isIndeterminate, value])
+const Checkbox = ({
+	id,
+	groupId = "",
+	name,
+	value = "",
+	label,
+	isLabelHidden = false,
+	isChecked = false,
+	isDisabled = false,
+	isSmall = false,
+	isIndeterminate = false,
+	onChange: propOnchange = (e: React.ChangeEvent<HTMLInputElement>) => {},
+}: CheckboxProps) => {
+	// Context for checkbox
+	const ctx = useCheckbox()
+	const [isHovered, isFocused, methods] = useInteraction({})
 
-		// Interaction methods for handling hover and focus
-		const [isHovered, isFocused, methods] = useInteraction({})
+	// Generate a dynamic ID for the checkbox
+	let uuid = `sui-checkbox-${useId()}`
 
-		// Generate a dynamic ID for the checkbox
-		let uuid = `sui-checkbox-${useId()}`
+	// use ID from props list if it exists
+	if (!!id) {
+		uuid = id
+	}
 
-		// use ID from props list if exists
-		if (!!id) {
-			uuid = id
-		}
+	useEffect(() => {
+		// Add the checkbox details to the context list on component mount
+		ctx?.addToList(id as string, isChecked, groupId)
```

Comment:
@abder can we fix this error ? 
![image](https://github.com/wpmudev/sui-react/assets/14994452/5d0acfc7-5966-4f94-9f06-d58f4eb193e5)